code/CloudFlare_Purge.php

```diff
@@ -0,0 +1,433 @@
+<?php
+
+class CloudFlare_Purge extends Object
+{
+
+    /**
+     * @var
+     */
+    protected $successMessage;
+
+    /**
+     * @var
+     */
+    protected $failureMessage;
+
+    /**
+     * @var array
+     */
+    protected $files;
+
+    /**
+     * @var bool
+     */
+    protected $purgeEverything = false;
+
+    /**
+     * @var
+     */
+    protected $response;
+
+    /**
+     * @var string
+     */
+    protected static $endpoint = "https://api.cloudflare.com/client/v4/zones/:identifier/purge_cache";
+
+    /**
+     * @param array $files
+     * @return $this
+     */
+    public function setFiles(array $files)
+    {
+        $this->clearFiles();
+        $this->pushFiles($files);
+
+        return $this;
+    }
+
+    /**
+     * @param string $file
+     * @return $this
+     */
+    public function pushFile($file)
+    {
+        if (!is_array($this->files)) {
+            $this->files = array();
+        }
+
+        if (is_array($file)) {
+            return $this->pushFiles($file);
+        }
+
+        array_push($this->files, $this->convertToAbsolute($file));
+
+        return $this;
+    }
+
+    /**
+     * @param $files
+     * @return CloudFlare_Purge
+     */
+    public function pushFiles($files)
+    {
+        if (is_string($files)) {
+            return $this->pushFile($files);
+        }
+
+        foreach ($files as $file) {
+            $this->pushFile($file);
+        }
+
+        return $this;
+    }
```

Comment:
You could probably do away with one of these two methods and just use one for everything